dragonchain/lib/dto/bnb.py

```diff
@@ -0,0 +1,319 @@
+            self.wallet = Wallet(private_key, env=prod_env)
+
+        self.priv_key = self.wallet.private_key
+        self.wallet_address = self.wallet.address
+
+    def ping(self) -> None:
+        """Ping this network to check if the given node is reachable and authorization is correct (raises exception if not)"""
+        self._call_node_rpc("status", {})
+
+    # https://docs.binance.org/api-reference/node-rpc.html#6114-query-tx
+    def is_transaction_confirmed(self, transaction_hash: str) -> bool:
+        """Check if a transaction is considered confirmed
+        Args:
+            transaction_hash: The hash (or equivalent, i.e. id) of the transaction to check
+        Returns:
+            Boolean if the transaction has received enough confirmations to be considered confirmed
+        Raises:
+            exceptions.RPCTransactionNotFound: When the transaction could not be found (may have been dropped)
+        """
+        _log.info(f"[BINANCE] Getting confirmations for {transaction_hash}")
+        transaction_hash = "0x" + transaction_hash  # GET call needs this in front of the hash
+        try:
+            response = self._call_node_rpc("tx", {"hash": transaction_hash, "prove": "true"})
+            transaction_block_number = response["result"]["height"]
+        except exceptions.RPCError:
+            raise exceptions.RPCTransactionNotFound(f"Transaction {transaction_hash} not found")
+        latest_block_number = self.get_current_block()
+        _log.info(f"[BINANCE] Latest block number: {latest_block_number} | Block number of transaction: {transaction_block_number}")
+        return transaction_block_number and (latest_block_number - transaction_block_number) >= CONFIRMATIONS_CONSIDERED_FINAL
+
+    # https://docs.binance.org/api-reference/api-server.html#apiv1balanceaddresssymbol
+    def check_balance(self, symbol: str = "BNB") -> int:
+        """Check the balance of the address for this network
+        Args:
+            symbol: stock ticker symbol for token you're trying to check the balance of
+        Returns:
+            The amount of funds of that token in the wallet
+        """
+        _log.info(f"[BNB] Checking {symbol} balance for {self.wallet_address}")
+        method = f"balances/{self.wallet_address}/{symbol}"
+        response = self._call_node_api(method, {})  # passing empty params dict
+        bnb_balance = response["balance"]["free"]
+        return bnb_balance
+
+    # https://docs.binance.org/api-reference/api-server.html#apiv1fees
+    def get_transaction_fee(self) -> int:
+        """Calculate the transaction fee estimate for a transaction given current fee rates
+        Returns:
+            The amount of estimated transaction fee cost for the network
+        """
+        _log.info("Double checking the send fee...")
+        response = self._call_node_api("fees", {})
+        fee_type = response[12]["fixed_fee_params"]["msg_type"]  # should be "send"
```

Comment:
don't hard code index of 12. Search for it so that we aren't caught off guard if stuff moves

Reply:
Fixed.